Ruins/src/main/java/atomicstryker/ruins/common/RuinTemplateRule.java

```diff
@@ -606,9 +606,12 @@ else if (dataString.startsWith("HardChest"))
         else if (dataString.startsWith("ChestGenHook:"))
         {
             String[] s = dataString.split(":");
+            if(s.length > 3){
+                s[1] = String.join(":", s[1], s[2]);
+                s[2] = s[3];
+            }
             int targetCount = s.length > 2 ? Integer.valueOf(s[s.length-1].split("-")[0]) : 0;

```

Comment:
If you're going to keep your change to use s[s.length-1] here instead of the original code's s[2], you can get rid of line 611. I added that line under the assumption nothing else was changing. Either way works, but you don't need both.

Reply:
Got it. Was treating the array as LinkedList lmao.